mcs/class/corlib/System.Threading/Timer.cs

```diff
@@ -33,8 +33,35 @@
 using System.Collections;
 using System.Runtime.CompilerServices;
 
+
 namespace System.Threading
 {
+#if WASM
+	internal static class WasmRuntime {
+		static Dictionary<int, Action> callbacks;
+		static int next_id;
+
+		[MethodImplAttribute(MethodImplOptions.InternalCall)]
+		static extern void SetTimeout (int timeout, int id);
+
+		internal static void ScheduleTimeout (int timeout, Action action) {
+			if (callbacks == null)
+				callbacks = new Dictionary<int, Action> ();
+			int id = ++next_id;
```

Comment:
Shouldn't this be atomic?

Reply:
There is no concurrency in WASM

Reply:
Yes, WebAssembly is single threaded for now. There is work to make it multi-threaded, but that's years away from having browsers shipping.